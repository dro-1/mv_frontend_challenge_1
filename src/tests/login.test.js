import React from 'react'
import { shallow } from 'enzyme'
import { LoginPage } from './../pages/login.page'

describe('Login Page', () => {
  it('correctly renders Login Page', () => {
    expect(shallow(<LoginPage />)).toMatchSnapshot()
  })

  it('calls setUser with the right parameters on login button click with the right login details', async () => {
    const mockSetUser = jest.fn()

    const loginWrapper = shallow(<LoginPage setUser={mockSetUser} />)

    loginWrapper
      .find('#password')
      .simulate('change', { target: { value: 'password' } })
    loginWrapper.find('#name').simulate('change', { target: { value: 'seun' } })
    loginWrapper.find('#login').simulate('click', {
      preventDefault: () => null,
    })

    expect(mockSetUser).toBeCalledWith({
      name: 'seun',
      balance: 0,
    })
  })

  it('does NOT call setUser if login button is clicked with wrong login details', async () => {
    const mockSetUser = jest.fn()

    const loginWrapper = shallow(<LoginPage setUser={mockSetUser} />)

    loginWrapper
      .find('#password')
      .simulate('change', { target: { value: 'pass' } })
    loginWrapper.find('#name').simulate('change', { target: { value: 'seun' } })
    loginWrapper.find('#login').simulate('click', {
      preventDefault: () => null,
    })

    expect(mockSetUser).toBeCalledTimes(0)
  })
})
