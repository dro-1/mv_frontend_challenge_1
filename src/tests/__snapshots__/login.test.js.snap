// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page correctly renders Login Page 1`] = `
<styled.div>
  <form>
    <header>
      <h1>
        Let's sign you in.
      </h1>
      <h2>
        Welcome back. You've been missed.
      </h2>
    </header>
    <input
      id="name"
      onChange={[Function]}
      placeholder="Username"
      type="text"
      value=""
    />
    <i>
      Username must not be empty
    </i>
    <input
      id="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
    <i>
      Password is password
    </i>
    <button
      id="login"
      onClick={[Function]}
    >
      Sign In
    </button>
  </form>
</styled.div>
`;
